packages/firestore/src/model/field_value.ts

```diff
@@ -634,6 +688,14 @@ export class ObjectValue extends FieldValue {
     return FieldMask.fromSet(fields);
   }
 
+  byteSize(): number {
+    let size = 0;
+    this.internalValue.inorderTraversal((key, val) => {
+      size += key.length + val.byteSize();
```

Comment:
For consistency with your other code, this would be 2* key.length, right?  But again, if we assume UTF-8 encoding, then this is okay.  

FWIW it is tempting to create a `Sizer` helper class that consistently deals with data types.  E.g. `Sizer.string(key.length) + val.byteSize()`